opal/mca/accelerator/cuda/configure.m4

```diff
@@ -15,11 +15,12 @@
 
 #
 # If CUDA support was requested, then build the CUDA support library.
-# This code checks just makes sure the check was done earlier by the
-# opal_check_cuda.m4 code.
-#
+# This code checks makes sure the check was done earlier by the
+# opal_check_cuda.m4 code. It also copies the flags and libs under
+# opal_cuda_CPPFLAGS, opal_cuda_LDFLAGS, and opal_cuda_LIBS
 
 AC_DEFUN([MCA_opal_accelerator_cuda_CONFIG],[
+
     AC_CONFIG_FILES([opal/mca/accelerator/cuda/Makefile])
 
     # make sure that CUDA-aware checks have been done
```

Comment:
you're a little indecisive on the interface for `OPAL_CHECK_CUDA`.  It's bad form to `AC_REQUIRE` a test that takes arguments.  You either need to update all the places that `AC_REQUIRE([OPAL_CHECK_CUDA])` to explicitly call `OPAL_CHECK_CUDA` with arguments, or have `OPAL_CHECK_CUDA` not take arguments.  My preference would be to explicitly call `OPAL_CHECK_CUDA([accelerator_cuda])` here, and then you don't need lines 33-35 below.

Reply:
Every test in the `OPAL_CHECK_CUDA` call should be cached between runs, so it should be very quick.  If you look, this is the approach taken by UCX or OFI or similar.

Reply:
ACK, I changed it